src/reducers/tasksReducer.js

```diff
@@ -32,6 +32,35 @@ export default function (state = initialState, action) {
     case 'POST_TASK_REJECTED':
       return state;
 
+    case 'DELETE_TASK_PENDING':
+      return state;
+
+    case 'DELETE_TASK_REJECTED':
+      return state;
+
+    case 'DELETE_TASK_FULFILLED':
+      const newState = { ...state };
```

Comment:
@Spetastian @MartyTheeMartian This creates a shallow copy, yes?  And thus the changes made below that modify the contents of `newState` also modify `state`, yes?  Or is this safe?

Reply:
Yes, this is short hand for object.assign. It will be a shallow copy of state.

Reply:
I would advice doing this:

```
return { 
 ...state,
 taskMap: state.taskMap.filter(t => *task not the one to be deleted*)
 rootTaskGids: state.rootTaskGids.filter(....)
```

Reply:
Is their a way to make a deep copy? I like this, but it excludes to removal of the task from it's parent and from `recentalyAddedTask`.